lib/chef/provider/package/chocolatey.rb

```diff
@@ -211,11 +211,18 @@ def args_to_string(*args)
         def available_packages
           @available_packages ||=
             begin
+              name = 0
+              version = 1
+
               cmd = [ "list -ar #{package_name_array.join ' '}" ]
               cmd.push( "-source #{new_resource.source}" ) if new_resource.source
-              parse_list_output(*cmd).reject do |name,version|
-                desired_name_versions[name] && desired_name_versions[name] != version
+              available = {}
+              parse_list_output(*cmd).each do |name_version|
+                if desired_name_versions[name_version[name]].nil? || desired_name_versions[name_version[name]] == name_version[version]
+                  available[name_version[name]] = name_version[version]
+                end
               end
+              available
```

Comment:
i suspect you can use a more functional notation (#each_with_object?) and get rid of the create-hash, loop-over-thing-populating-hash, return hash pattern.